Bottom: 7d79263f667821c52ff6013d994014843cb55bb2
Top:    7767f27c45c90faf1ae234273ee676830b2d6b3b
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-03-05 16:31:27 +0000

Add void *userdata member to the POLYAREA object

This is intended to allow client code operating on polygons to assign
arbitrary additional data to the polygon contours.

The initial target use-case is in 3D export code, where it is helpful to
have temporary links between POLYAREA contours being extruded and the 3D
object structures being generated. Allowing this tagging directly avoids
the need to maintain a hash-mapping or similar.



---

diff --git a/src/polyarea.h b/src/polyarea.h
index 76b724a..bcb2152 100644
--- a/src/polyarea.h
+++ b/src/polyarea.h
@@ -159,6 +159,7 @@ struct POLYAREA
     PLINE *contours;
     rtree_t *contour_tree;
     POLYPARENTAGE parentage;
+    void *user_data; /* To be used at will by the owner of this polygon */
 };
 
 void poly_dump (POLYAREA *p);
diff --git a/src/polygon1.c b/src/polygon1.c
index 2008916..8041c05 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -3860,6 +3860,7 @@ poly_Init (POLYAREA * p)
   p->contours = NULL;
   p->contour_tree = r_create_tree (NULL, 0, 0);
   p->parentage = no_parentage;
+  p->user_data = NULL;
 }
 
 POLYAREA *
